misk-hibernate/src/main/kotlin/misk/hibernate/RealTransacter.kt

```diff
@@ -195,39 +215,75 @@ internal class RealTransacter private constructor(
     }
   }
 
-  override fun retries(maxAttempts: Int): Transacter = withOptions(
-      options.copy(maxAttempts = maxAttempts))
+  override fun retries(maxAttempts: Int): Transacter =
+      withOptions(options.copy(maxAttempts = maxAttempts))
 
   override fun allowCowrites(): Transacter {
     val disableChecks = options.disabledChecks.clone()
     disableChecks.add(Check.COWRITE)
-    return withOptions(
-        options.copy(disabledChecks = disableChecks))
+    return withOptions(options.copy(disabledChecks = disableChecks))
   }
 
   override fun noRetries(): Transacter = withOptions(options.copy(maxAttempts = 1))
 
+  internal fun withTimeouts(timeouts: TimeoutsConfig): Transacter {
+    return RealTransacter(
+        qualifier = qualifier,
+        sessionFactoryProvider = sessionFactoryProvider,
+        threadLatestSession = threadLatestSession,
+        options = options,
+        queryTracingListener = queryTracingListener,
+        tracer = tracer,
+        transactionRunnerFactory = transactionRunnerFactory,
+        querySniperFactory = querySniperFactory,
+        timeoutsConfig = timeouts
+    )
+  }
+
+  override fun withTimeouts(timeouts: Timeouts): Transacter {
+    return withTimeouts(timeoutsConfig.copy(transaction = timeouts))
+  }
+
+  override fun withQueryTimeouts(timeouts: Timeouts): Transacter {
+    return withTimeouts(timeoutsConfig.copy(query = timeouts))
+  }
+
+  override fun withSlowQueryTimeouts(timeouts: Timeouts): Transacter {
+    return withTimeouts(timeoutsConfig.copy(slowQuery = timeouts))
+  }
+
+  override fun noTimeouts(): Transacter = withOptions(options.copy(disableTimeouts = true))
```

Comment:
This won't work on Vitess. Vitess always enforces query and transaction timeouts.